Source/Foundation/bsfUtility/Error/BsCrashHandler.h

```diff
@@ -2,11 +2,37 @@
 //*********** Licensed under the MIT license. See LICENSE.md for full terms. This notice is not to be removed. ***********//
 #pragma once
 
+#include <functional>
+
 #define BS_MAX_STACKTRACE_DEPTH 200
 #define BS_MAX_STACKTRACE_NAME_BYTES 1024
 
 namespace bs
 {
+	/** Struct for holding crash handler settings */
+	struct CrashHandlerSettings
+	{
+		/** Called at the start of a when a windows SEH exception is started to be handled. Return true to skip default action */
+		std::function<bool(const String& type,
+			const String& description,
+			const String& function,
+			const String& file,
+			UINT32 line)> onBeforeReportCrash;
+
+#if BS_PLATFORM == BS_PLATFORM_WIN32
+		/** Called at the start when a windows SEH exception is started to be handled. Return true to skip default action */
+		std::function<bool(void* exceptionData)> onBeforeWindowsSEHReportCrash;
+#endif
+		/**
+		* Called after the crash callstack is written to log. Return true to skip writing to file and doing
+		* further other on crash actions
+		*/
+		std::function<bool()> onCrashPrintedToLog;
```

Comment:
I'd prefer this be named onAfterReportCrash, to be more consistent with the naming + printing to log might not happen, we might just save a dump file, in which case its inacurrate.

Reply:
I don't want BSF to actually write any log files. So I want this callback to be between the print to log and save log to file calls in the crash handler. The name change wouldn't completely capture the entire meaning.

Reply:
Makes sense.